src/renderers/dom/client/ReactBrowserEventEmitter.js

```diff
@@ -328,6 +369,14 @@ var ReactBrowserEventEmitter = Object.assign({}, ReactEventEmitterMixin, {
         isListening[dependency] = true;
       }
     }
+
+    return function() {
```

Comment:
Quickly reading this I'm having a little trouble understanding the exact flow, if I understand this correctly, this seems like a wasteful way of doing it. Allocating a new callback which holds an array of callbacks, instead of keeping a single global array of callbacks? Or am I missing something.